reactos/drivers/usb/usbohci_new/roothub.c

```diff
@@ -0,0 +1,585 @@
+
+    return MP_STATUS_SUCCESS;
+}
+
+MPSTATUS
+NTAPI
+OHCI_RH_ClearFeaturePortConnectChange(IN PVOID ohciExtension,
+                                      IN USHORT Port)
+{
+    POHCI_EXTENSION OhciExtension;
+    POHCI_OPERATIONAL_REGISTERS OperationalRegs;
+    PULONG PortStatusReg;
+    OHCI_REG_RH_PORT_STATUS PortStatus;
+
+    OhciExtension = ohciExtension;
+
+    DPRINT("OHCI_RH_ClearFeaturePortConnectChange: OhciExtension - %p, Port - %x\n",
+           OhciExtension,
+           Port);
+
+    ASSERT(Port > 0);
+
+    OperationalRegs = OhciExtension->OperationalRegs;
+    PortStatusReg = (PULONG)&OperationalRegs->HcRhPortStatus[Port-1];
+
+    PortStatus.AsULONG = 0;
+    PortStatus.ConnectStatusChange = 1;
+
+    WRITE_REGISTER_ULONG(PortStatusReg, PortStatus.AsULONG);
+
+    return MP_STATUS_SUCCESS;
+}
+
+MPSTATUS
+NTAPI
+OHCI_RH_ClearFeaturePortResetChange(IN PVOID ohciExtension,
+                                    IN USHORT Port)
+{
+    POHCI_EXTENSION OhciExtension;
+    POHCI_OPERATIONAL_REGISTERS OperationalRegs;
+    PULONG PortStatusReg;
+    OHCI_REG_RH_PORT_STATUS PortStatus;
+
+    OhciExtension = ohciExtension;
+
+    DPRINT("OHCI_RH_ClearFeaturePortResetChange: OhciExtension - %p, Port - %x\n",
+           OhciExtension,
+           Port);
+
+    ASSERT(Port > 0);
+
+    OperationalRegs = OhciExtension->OperationalRegs;
+    PortStatusReg = (PULONG)&OperationalRegs->HcRhPortStatus[Port-1];
+
+    PortStatus.AsULONG = 0;
+    PortStatus.PortResetStatusChange = 1;
+
+    WRITE_REGISTER_ULONG(PortStatusReg, PortStatus.AsULONG);
+
+    return MP_STATUS_SUCCESS;
+}
+
+MPSTATUS
+NTAPI
+OHCI_RH_ClearFeaturePortSuspendChange(IN PVOID ohciExtension,
+                                      IN USHORT Port)
+{
+    POHCI_EXTENSION OhciExtension;
+    POHCI_OPERATIONAL_REGISTERS OperationalRegs;
+    PULONG PortStatusReg;
+    OHCI_REG_RH_PORT_STATUS PortStatus;
+
+    OhciExtension = ohciExtension;
+
+    DPRINT("OHCI_RH_ClearFeaturePortSuspendChange: OhciExtension - %p, Port - %x\n",
+           OhciExtension,
+           Port);
+
+    ASSERT(Port > 0);
+
+    OperationalRegs = OhciExtension->OperationalRegs;
+    PortStatusReg = (PULONG)&OperationalRegs->HcRhPortStatus[Port-1];
+
+    PortStatus.AsULONG = 0;
+    PortStatus.PortSuspendStatusChange = 1;
+
+    WRITE_REGISTER_ULONG(PortStatusReg, PortStatus.AsULONG);
+
+    return MP_STATUS_SUCCESS;
+}
+
+MPSTATUS
+NTAPI
+OHCI_RH_ClearFeaturePortOvercurrentChange(IN PVOID ohciExtension,
+                                          IN USHORT Port)
+{
+    POHCI_EXTENSION OhciExtension;
+    POHCI_OPERATIONAL_REGISTERS OperationalRegs;
+    PULONG PortStatusReg;
+    PULONG RhStatusReg;
+    OHCI_REG_RH_PORT_STATUS PortStatus;
+    OHCI_REG_RH_STATUS RhStatus;
+
+    OhciExtension = ohciExtension;
+
+    DPRINT("OHCI_RH_ClearFeaturePortOvercurrentChange: OhciExtension - %p, Port - %x\n",
+           OhciExtension,
+           Port);
+
+    ASSERT(Port >= 0);
```

Comment:
The issue here got fixed by moving the StatusReg assignment inside each if block. The assert is not needed (and is a tautology since USHORT is always >= 0)

Reply:
Ok, deleted.